examples/vision/yolov8.py

```diff
@@ -0,0 +1,628 @@
+define a bounding box:  xmin, ymin, xmax, ymax.
+
+The `'classes'` Tensor has a shape of `[batch, num_boxes]`, where each element represents
+the class label for the corresponding bounding box in the `'boxes'` Tensor. The num_boxes
+dimension may be ragged, which means that the number of boxes may vary across images in
+the batch.
+
+Final dict should be:
+```python
+{"images": images, "bounding_boxes": bounding_boxes}
+```
+"""
+
+
+def load_image(image_path):
+    image = tf.io.read_file(image_path)
+    image = tf.image.decode_jpeg(image, channels=3)
+    return image
+
+
+def load_dataset(image_path, classes, bbox):
+    # Read Image
+    image = load_image(image_path)
+    bounding_boxes = {
+        "classes": tf.cast(classes, dtype=tf.float32),
+        "boxes": bbox,
+    }
+    return {"images": tf.cast(image, tf.float32), "bounding_boxes": bounding_boxes}
+
+
+"""
+Here we create a layer that resizes images to 640x640 pixels, while maintaining the
+original aspect ratio. The bounding boxes associated with the image are specified in the
+`xyxy` format. If necessary, the resized image will be padded with zeros to maintain the
+original aspect ratio.
+
+Bounding Box Formats supported by KerasCV:
+1.   CENTER_XYWH
+2.   XYWH
+3.   XYXY
+4.   REL_XYXY
+5.   REL_XYWH 
+6.   YXYX
+7.   REL_YXYX
+
+
+You can read more about KerasCV bounding box formats in
+[docs](https://keras.io/api/keras_cv/bounding_box/formats/).
+
+Furthermore, it is possible to perform format conversion between any two pairs:
+
+```python
+boxes = keras_cv.bounding_box.convert_format(
+        bounding_box,
+        images=image,
+        source="xyxy",  # Original Format
+        target="xywh",  # Target Format (to which we want to convert)
+    )
+```
+"""
+
+"""
+## Data Augmentation
+
+One of the most challenging tasks when constructing object detection pipelines is data
+augmentation. It involves applying various transformations to the input images to
+increase the diversity of the training data and improve the model's ability to
+generalize. However, when working with object detection tasks, it becomes even more
+complex as these transformations need to be aware of the underlying bounding boxes and
+update them accordingly.
+
+KerasCV provides native support for bounding box augmentation. KerasCV offers an
+extensive collection of data augmentation layers specifically designed to handle bounding
+boxes. These layers intelligently adjust the bounding box coordinates as the image is
+transformed, ensuring that the bounding boxes remain accurate and aligned with the
+augmented images.
+
+By leveraging KerasCV's capabilities, developers can conveniently integrate bounding
+box-friendly data augmentation into their object detection pipelines. By performing
+on-the-fly augmentation within a tf.data pipeline, the process becomes seamless and
+efficient, enabling better training and more accurate object detection results.
+"""
+
+augmenter = keras.Sequential(
```

Comment:
Mosaic and MixUp must be used in data augmentation pipeline. Very important component for Yolo model (v5, v8).

```python
keras_cv.layers.RandomChoice(
  layers=[
      keras_cv.layers.Mosaic(bounding_box_format="xywh"),
      keras_cv.layers.MixUp(alpha=0.4)
     ]
),
```

( However, [issue](https://github.com/keras-team/keras-cv/issues/1782) with random-choice. But only one can be used with random-choice layer. )

Reply:
Yeah, I will add this!